src/main/java/com/github/holly/accountability/password_reset_email/PasswordEmailController.java

```diff
@@ -0,0 +1,152 @@
+package com.github.holly.accountability.password_reset_email;
+
+import com.github.holly.accountability.config.GenericResponse;
+import com.github.holly.accountability.config.properties.ApplicationProperties;
+import jakarta.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.FieldError;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+
+
+@Controller
+@RequestMapping("/email")
+public class PasswordEmailController {
+
+    private final ApplicationProperties applicationProperties;
+    //normally encode parameters or when you're sending a URL to the browser somewhere
+    public static final String CHANGE_PASSWORD_FROM_TOKEN = "/change-password-from-token";
+
+    private final PasswordEmailService passwordEmailService;
+
+    @Autowired
+    public PasswordEmailController(PasswordEmailService passwordEmailService,
+                                   ApplicationProperties applicationProperties) {
+        this.passwordEmailService = passwordEmailService;
+        this.applicationProperties = applicationProperties;
+    }
+
+    @ResponseBody
+    @GetMapping("/send-token")
+    public GenericResponse sendPasswordEmail(@RequestParam String email) {
+        try {
+            return passwordEmailService.sendPasswordEmail(email);
+        } catch (Exception e) {
+            return new GenericResponse(e.getMessage());
+        }
+    }
+
+    @GetMapping(CHANGE_PASSWORD_FROM_TOKEN + "/{token}")
+    public ResponseEntity<?>  showChangePasswordFromTokenPage(@PathVariable("token") String token) {
```

Comment:
This is a great use of ResponseEntity to handle both an error case and expected success case via redirect.